Author: Guillaume Delacour <gui@iroqwa.org>
Description: Use pkg-config for imagemagick version detection
 As Wand-config is no more available since Debian imagemagick 8:6.8.9.6-4
 (#761927)
Last-Update: 2014-10-09

Index: php-imagick.git/imagick-3.2.0RC1/imagemagick.m4
===================================================================
--- php-imagick.git.orig/imagick-3.2.0RC1/imagemagick.m4
+++ php-imagick.git/imagick-3.2.0RC1/imagemagick.m4
@@ -5,7 +5,6 @@
 # bin/MagickWand-config
 #
 # Sets
-#  IM_WAND_BINARY
 #  IM_IMAGEMAGICK_PREFIX
 #  IM_IMAGEMAGICK_VERSION
 #  IM_IMAGEMAGICK_VERSION_MASK
@@ -35,49 +34,12 @@ AC_DEFUN([IM_FIND_IMAGEMAGICK],[
 
   AC_MSG_CHECKING(ImageMagick MagickWand API configuration program)
 
-  if test "$IM_EXTRA_SEARCH_PREFIX" != "yes"; then
-    for i in "$IM_EXTRA_SEARCH_PREFIX" /usr/local /usr /opt /opt/local;
-    do
-      if test -r "${i}/bin/MagickWand-config"; then
-        IM_WAND_BINARY="${i}/bin/MagickWand-config"
-        IM_IMAGEMAGICK_PREFIX=$i
-        break
-      fi
-
-      if test -r "${i}/bin/Wand-config"; then
-        IM_WAND_BINARY="${i}/bin/Wand-config"
-        IM_IMAGEMAGICK_PREFIX=$i
-        break
-      fi
-    done
-  else
-    for i in /usr/local /usr /opt /opt/local;
-    do
-      if test -r "${i}/bin/MagickWand-config"; then
-        IM_WAND_BINARY="${i}/bin/MagickWand-config"
-        IM_IMAGEMAGICK_PREFIX=$i
-        break
-      fi
-
-      if test -r "${i}/bin/Wand-config"; then
-        IM_WAND_BINARY="${i}/bin/Wand-config"
-        IM_IMAGEMAGICK_PREFIX=$i
-        break
-      fi
-    done
-  fi
-
-  if test "x" = "x$IM_WAND_BINARY"; then
-    AC_MSG_ERROR(not found. Please provide a path to MagickWand-config or Wand-config program.)
-  fi
-  AC_MSG_RESULT([found in $IM_WAND_BINARY])
-
 # This is used later for cflags and libs
   export PKG_CONFIG_PATH="${IM_IMAGEMAGICK_PREFIX}/${PHP_LIBDIR}/pkgconfig"
   
 # Check version
 #
-  IM_IMAGEMAGICK_VERSION=`$IM_WAND_BINARY --version`
+  IM_IMAGEMAGICK_VERSION=`$PKG_CONFIG --modversion MagickWand`
   IM_IMAGEMAGICK_VERSION_MASK=`echo $IM_IMAGEMAGICK_VERSION | $AWK 'BEGIN { FS = "."; } { printf "%d", ($[1] * 1000 + $[2]) * 1000 + $[3];}'`
 
   IM_MIMIMUM_VERSION_MASK=`echo $IM_MINIMUM_VERSION | $AWK 'BEGIN { FS = "."; } { printf "%d", ($[1] * 1000 + $[2]) * 1000 + $[3];}'`
@@ -97,7 +59,7 @@ AC_DEFUN([IM_FIND_IMAGEMAGICK],[
 
   AC_MSG_CHECKING(for MagickWand.h or magick-wand.h header)
 
-  IM_PREFIX=`$IM_WAND_BINARY --prefix`
+  IM_PREFIX=`$PKG_CONFIG --variable=prefix MagickWand`
   IM_MAJOR_VERSION=`echo $IM_IMAGEMAGICK_VERSION | $AWK 'BEGIN { FS = "."; } {print $[1]}'`
 
   # Try the header formats from newest to oldest
@@ -140,11 +102,10 @@ AC_DEFUN([IM_FIND_IMAGEMAGICK],[
 #
 # The cflags and libs
 #
-  IM_IMAGEMAGICK_LIBS=`$IM_WAND_BINARY --libs`
-  IM_IMAGEMAGICK_CFLAGS=`$IM_WAND_BINARY --cflags`
+  IM_IMAGEMAGICK_LIBS=`$PKG_CONFIG --libs MagickWand`
+  IM_IMAGEMAGICK_CFLAGS=`$PKG_CONFIG --cflags MagickWand`
 
   export IM_IMAGEMAGICK_PREFIX
-  export IM_WAND_BINARY
   export IM_IMAGEMAGICK_VERSION
   export IM_IMAGEMAGICK_VERSION_MASK
   export IM_INCLUDE_FORMAT
